From: REG-Linux <cjom@reglinux.org>
Subject: [PATCH] cave/cv1k: restore Akai Katana and DoDonPachi SaiDaiOuJou

Both drivers were removed upstream at the rightholder's request; re-enable
them and list them in mame.lst.

Refreshed for GroovyMAME 0.289. Note the ROM_START blocks are no longer
commented out line by line as in 0.287 -- they are live text wrapped in an
"#if 0", so the guard has to go too or the GAME() lines fail to compile with
"'rom_akatana' was not declared in this scope".

--- a/src/mame/cave/cv1k.cpp	2026-08-14 00:20:16.311139594 +0100
+++ b/src/mame/cave/cv1k.cpp	2026-08-14 00:20:37.700106502 +0100
@@ -188,10 +188,6 @@
  - Experimental SH7709S cache/memory timing
  - Requires to measure screen raw params for correct video timing?
 
-Removed games
- - 31/12/2021, Akai Katana and Dodonpachi Saidaioujou were removed at the request of the current
-   rightholder, exA-Arcadia (legal@exa.ac).
-
 */
 
 #include "emu.h"
@@ -975,7 +971,6 @@
 	ROM_LOAD16_WORD_SWAP( "u24", 0x400000, 0x400000, CRC(31f9eb0a) SHA1(322158779e969bb321241065dd49c1167b91ff6c) )
 ROM_END
 
-#if 0 // not included in MAME, see driver notes
 ROM_START( akatana )
 	ROM_REGION( 0x400000, "maincpu", ROMREGION_ERASEFF)
 	ROM_LOAD16_WORD_SWAP( "u4", 0x000000, 0x400000, CRC(613fd380) SHA1(6e28480eef3b483d00b42d811a9d2c7fa1097924) ) // (2010/ 8/13 MASTER VER.)
@@ -1002,7 +997,6 @@
 	ROM_LOAD16_WORD_SWAP("u24", 0x400000, 0x400000, CRC(f593045b) SHA1(91b92870d0dd2a7817cb0059cc750e2393686f4c) )
 	ROM_IGNORE( 0x000100 ) // Flash extra bytes
 ROM_END
-#endif
 
 u64 cv1k_state::speedup_r()
 {
@@ -1128,10 +1122,10 @@
 GAME( 2010, dfkbl,      0,        cv1k_d, cv1k, cv1k_state, init_ddpdfk,   ROT270, "Cave",                 "DoDonPachi Dai-Fukkatsu Black Label (Japan, 2010/1/18 BLACK LABEL)",                   MACHINE_IMPERFECT_TIMING )
 
 // CA021  Akai Katana
-//GAME( 2010, akatana,    0,        cv1k_d, cv1k, cv1k_state, init_ddpdfk,   ROT0,   "Cave",                 "Akai Katana (Japan, 2010/ 8/13 MASTER VER.)",                                          MACHINE_IMPERFECT_TIMING )
+GAME( 2010, akatana,    0,        cv1k_d, cv1k, cv1k_state, init_ddpdfk,   ROT0,   "Cave",                 "Akai Katana (Japan, 2010/ 8/13 MASTER VER.)",                                          MACHINE_IMPERFECT_TIMING )
 
 // CA??? DoDonPachi SaiDaiOuJou
-//GAME( 2012, ddpsdoj,    0,        cv1k_d, cv1k, cv1k_state, init_ddpdfk,   ROT270, "Cave",                 "DoDonPachi SaiDaiOuJou (Japan, 2012/ 4/20)",                                           MACHINE_IMPERFECT_TIMING )
+GAME( 2012, ddpsdoj,    0,        cv1k_d, cv1k, cv1k_state, init_ddpdfk,   ROT270, "Cave",                 "DoDonPachi SaiDaiOuJou (Japan, 2012/ 4/20)",                                           MACHINE_IMPERFECT_TIMING )
 
 // CMDL01 Medal Mahjong Moukari Bancho
 GAME( 2007, mmmbanc,    0,        cv1k,   mmmbanc, cv1k_state, init_pinkswts, ROT0,   "Cave (AMI license)",   "Medal Mahjong Moukari Bancho (Japan, 2007/06/05 MASTER VER.)",                         MACHINE_NOT_WORKING )
--- a/src/mame/mame.lst	2026-08-13 21:26:31.818465064 +0100
+++ b/src/mame/mame.lst	2026-08-13 21:26:31.838337495 +0100
@@ -16951,6 +16951,8 @@
 pinkswtsb
 pinkswtsx
 pinkswtssc
+akatana                         // (c) 2010 Cave (AMI license) - 2010/ 8/13 MASTER VER.
+ddpsdoj                         // (c) 2012 Cave - 2012/04/20
 
 @source:cave/fstgfish.cpp
 fstgfish
